server/php-queries.h

```diff
@@ -18,51 +19,46 @@ struct FinishedJob;
 struct JobSharedMessage;
 } // namespace job_workers
 
-enum class net_event_type_t {
-  rpc_answer,
-  rpc_error,
-  job_worker_answer,
+namespace net_events_data {
+
+struct rpc_answer {
+  int result_len;
+  //allocated via dl_malloc
+  char *result;
+};
+
+struct rpc_error {
+  int error_code;
+  const char *error_message;
+};
+
+struct job_worker_answer {
+  job_workers::FinishedJob *job_result;
 };
 
+} // namespace net_events_data
+
 struct net_event_t {
-  net_event_type_t type;
-  union {
-    slot_id_t slot_id;
-    slot_id_t rpc_id;
-  };
-  union {
-    struct { //rpc_answer
-      int result_len;
-      //allocated via dl_malloc
-      char *result;
-    };
-    struct { //rpc_error
-      int error_code;
-      const char *error_message;
-    };
-    struct { // job_worker_answer
-      job_workers::FinishedJob *job_result;
-    };
-  };
+  slot_id_t slot_id;
+  std::variant<net_events_data::rpc_answer, net_events_data::rpc_error, net_events_data::job_worker_answer> data;
 
   const char *get_description() const noexcept;
 };
 
-enum net_query_type_t {
-  nq_rpc_send
+namespace net_queries_data {
```

Comment:
namespace with only one type looks a bit strange, maybe combine it with another one above? Or remove it all?

Reply:
Second type will be added in this PR: https://github.com/VKCOM/kphp/pull/393

Reply:
got it